mesonbuild/modules/python.py

```diff
@@ -443,13 +371,12 @@ def _check_version(self, version: str) -> bool:
 
     def sanity(self, state: T.Optional['ModuleState'] = None) -> bool:
         # Sanity check, we expect to have something that at least quacks in tune
-        from tempfile import NamedTemporaryFile
-        with NamedTemporaryFile(suffix='.py', delete=False, mode='w', encoding='utf-8') as tf:
-            tmpfilename = tf.name
-            tf.write(INTROSPECT_COMMAND)
-        cmd = self.get_command() + [tmpfilename]
-        p, stdout, stderr = mesonlib.Popen_safe(cmd)
-        os.unlink(tmpfilename)
+
+        import importlib.resources
+
+        with importlib.resources.path('mesonbuild.scripts', 'python_info.py') as f:
+            cmd = self.get_command() + [f]
+            p, stdout, stderr = mesonlib.Popen_safe(cmd)
```

Comment:
`importlib.resources.path()` returns `pathlib.Path` object. The `subprocess` modules (used under the hood in `mesonlib.Popen_safe()`) accepts path objects as part of the command line only starting with Python 3.8. This should probably be
```python
cmd = self.get_command() + [os.fspath(f)]
```
or something similar.